### PR TITLE
Re: Create new `media` module

### PR DESCRIPTION
**Pull request by imashnake0**
[comment]: # (Replace [ ] with [x].)
- [x] Read [`CONTRIBUTING.md`](https://github.com/imashnake0/Animite/blob/be53ba4561c8ff20f588fb348965f208e301b6b8/CONTRIBUTING.md).

Reopening #212. Created a new media module since we need this code in the profile module (for navigating to media pages) => circular dependency (app <-> profile).

**Summary of changes:**

1. Created a new media module.
2. Moved media files from app to new media module.

**Tested changes:**

i think everything still works as expected

[comment]: # (THANK YOU! ♡\(灬♥ ◡ ♥灬\))

### CODE REVIEW

File: app/src/main/kotlin/com/imashnake/animite/features/home/HomeScreen.kt [hunk 1]
```diff
@@ -120,7 +119,10 @@ fun HomeScreen(
             }
         } while (true)
     }
-    val shader = remember { RuntimeShader(etherealShader) }
+    val shader = remember {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU)
```
Comment (boswelja) [inline app/src/main/kotlin/com/imashnake/animite/features/home/HomeScreen.kt:123]: What's going on here

File: app/src/main/kotlin/com/imashnake/animite/features/home/HomeScreen.kt [hunk 2]
```diff
@@ -77,17 +76,17 @@ import com.imashnake.animite.core.ui.ProgressIndicator
...
-import com.imashnake.animite.features.media.MediaPage
+import com.imashnake.animite.media.MediaPage
 import com.imashnake.animite.navigation.SharedContentKey
 import com.imashnake.animite.navigation.SharedContentKey.Component.Card
 import com.imashnake.animite.navigation.SharedContentKey.Component.Image
 import com.imashnake.animite.navigation.SharedContentKey.Component.Page
 import com.imashnake.animite.navigation.SharedContentKey.Component.Text
 import com.materialkolor.ktx.hasEnoughContrast
 import com.imashnake.animite.core.R as coreR
+import com.imashnake.animite.media.R as mediaR
```
Comment (boswelja) [inline app/src/main/kotlin/com/imashnake/animite/features/home/HomeScreen.kt:87]: Sad R noises

File: app/src/main/kotlin/com/imashnake/animite/features/home/HomeScreen.kt [hunk 3]
```diff
@@ -144,15 +146,24 @@ fun HomeScreen(
...
-                                        with(shader) {
-                                            setFloatUniform("resolution", size.width, size.height)
-                                            setFloatUniform("time", time.floatValue)
-                                            setColorUniform("orb", Color(0xFF6C408D).toArgb())
-                                            setColorUniform("bg", android.graphics.Color.TRANSPARENT)
-                                        }
-                                        onDrawBehind {
-                                            drawRect(ShaderBrush(shader))
-                                        }
+                                        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU) {
```
Comment (boswelja) [inline app/src/main/kotlin/com/imashnake/animite/features/home/HomeScreen.kt:149]: Can't just `if shader != null` here?

File: app/src/main/kotlin/com/imashnake/animite/features/home/HomeScreen.kt (hunk 1 above)
Comment (imashnake0) [reply app/src/main/kotlin/com/imashnake/animite/features/home/HomeScreen.kt:123]: This shader stuff requires tiramisu 😔

File: app/src/main/kotlin/com/imashnake/animite/features/home/HomeScreen.kt (hunk 3 above)
Comment (imashnake0) [reply app/src/main/kotlin/com/imashnake/animite/features/home/HomeScreen.kt:149]: No, we have to return `DrawResult` in `drawWithCache`

File: app/src/main/kotlin/com/imashnake/animite/features/home/HomeScreen.kt (hunk 1 above)
Comment (boswelja) [reply app/src/main/kotlin/com/imashnake/animite/features/home/HomeScreen.kt:123]: So why wasn't it in the shader PR 😔

File: app/src/main/kotlin/com/imashnake/animite/features/home/HomeScreen.kt (hunk 3 above)
Comment (boswelja) [reply app/src/main/kotlin/com/imashnake/animite/features/home/HomeScreen.kt:149]: I mean we have an if/else here already, I was wondering if we can replace that with a check on `shader` instead

File: app/src/main/kotlin/com/imashnake/animite/features/home/HomeScreen.kt (hunk 1 above)
Comment (imashnake0) [reply app/src/main/kotlin/com/imashnake/animite/features/home/HomeScreen.kt:123]: forgor 😔

**Pull request merged**